internal/url/url.go

```diff
@@ -35,18 +34,60 @@ func ParseQuery(query string, separator string) (map[string][]string, error) {
 			continue
 		}
 		value := ""
-		if i := strings.Index(key, "="); i >= 0 {
+		if i := strings.IndexByte(key, '='); i >= 0 {
 			key, value = key[:i], key[i+1:]
 		}
-		key, err := url.QueryUnescape(key)
-		if err != nil {
-			return nil, err
+		key = QueryUnescape(key)
+		value = QueryUnescape(value)
+		m[key] = append(m[key], value)
+	}
+	return m
+}
+
+func QueryUnescape(input string) string {
```

Comment:
Should this method be private? I don't see it being used outside of this package.

Reply:
Also, we definitively need to cover this method, did we copied and modified from somewhere? If that is the case we definitively need to reference it.

Reply:
We should unexport it and add more tests. It's based on https://github.com/SpiderLabs/ModSecurity/blob/4127c1bf52d2b30a5c2c3e641b8085fd9a720f43/src/utils/decode.cc#L25